pkg/controller/volume/expand/config/v1alpha1/defaults.go

```diff
@@ -0,0 +1,40 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubectrlmgrconfigv1alpha1 "k8s.io/kube-controller-manager/config/v1alpha1"
+)
+
+// RecommendedDefaultExpandControllerConfiguration defaults a pointer to a
+// AttachDetachControllerConfiguration struct. This will set the recommended default
+// values, but they may be subject to change between API versions. This function
+// is intentionally not registered in the scheme as a "normal" `SetDefaults_Foo`
```

Comment:
config structs that are only used in one context should set actual defaults. the "RecommendedDefaults" helper function pattern was for common structs like client config, leader election, etc, which are used in multiple contexts.

Reply:
I have  changed the file a lot. However, I found some controllers liker pv controller, ad controller, or hpa controller also use it to recommend some parameters in the config structs

Reply:
Yes, I opened https://github.com/kubernetes/kubernetes/issues/76027 to track fixing those

Reply:
Great! Can you help review the other parts of this PR?